sklearn/cross_decomposition/pls_.py

```diff
@@ -375,14 +375,14 @@ def transform(self, X, Y=None, copy=True):
         x_scores if Y is not given, (x_scores, y_scores) otherwise.
         """
         check_is_fitted(self, 'x_mean_')
-        X = check_array(X, copy=copy)
+        X = check_array(X, copy=copy, dtype=np.float64)
```

Comment:
> Sorry looks like dtype=FLOAT_DTYPES prefers np.float64 over np.float32!

I think that sounds right: converting an int to a float should prefer
float64, unless there is specific reasons.

I am less worried about such a conversion, as in a sense the user is not
being careful with his or her types. I am worried by a conversion from
float32 to float64, as float32 could very well be legit.